src/extensions/registries/protocol-handler-registry.ts

```diff
@@ -0,0 +1,10 @@
+import { RouteHandler } from "../../common/protocol-handler";
+
+/**
+ * ProtocolHandlerRegistration is the data required for an extension to register
+ * a handler to a specific path or dynamic path.
+ */
+export interface ProtocolHandlerRegistration {
+  pathSchema: string;
```

Comment:
Seems a bit over the top to name it "pathSchema". How about just "path"?

Reply:
Why is it over the top? it is a schema in so far as you can describe the shape of a URL.

For instance this is one that I use internally:

```
/:extPubName(\@[A-Za-z0-9_]+)?/:extName`
```

which defines a two part segment where the first must match the regex `/@[A-Za-z0-9_]+/` but is also optional.

Reply:
I get why it's not _just_ a plain path and there's a schema to it, but react-router, for example, still just calls it a "path": `<Route path="/users/profile/:id">` Just seems simpler in terms of naming. I thought I'd point that out as a comparison.